packages/-ember-data/tests/integration/adapter/rest-adapter-test.js

```diff
@@ -2137,697 +2242,688 @@ module('integration/adapter/rest_adapter - REST Adapter', function(hooks) {
+          await assert.expectWarning(async () => {
+            return post.save().then(() => assert.ok(true, 'save fullfills correctly'));
+          }, /JSON/);
+        }
+      );
 
-  test('error handling with a very long HTML-formatted payload truncates the friendly message', assert => {
-    assert.expect(2);
+      test('can return an empty 200 response, though a valid stringified JSON is expected', async function(assert) {
+        assert.expect(1);
 
-    ajaxError(new Array(100).join('<blink />'), 500, { 'Content-Type': 'text/html' });
+        server.put('/posts/1', function() {
+          return [200, { 'Content-Type': 'application/json' }, ''];
+        });
 
-    run(() => {
-      store.findRecord('post', '1').catch(err => {
-        assert.equal(
-          err.message,
-          'Ember Data Request GET /posts/1 returned a 500\nPayload (text/html)\n[Omitted Lengthy HTML]'
-        );
-        assert.ok(err, 'promise rejected');
+        let post = store.push({ data: { id: '1', type: 'post' } });
+        return post.save().then(() => assert.ok(true, 'save fullfills correctly'));
       });
-    });
-  });
 
-  test('findAll resolves with a collection of DS.Models, not DS.InternalModels', assert => {
-    assert.expect(4);
+      test('can return a null 200 response, though a valid stringified JSON is expected', async function(assert) {
+        assert.expect(1);
 
-    ajaxResponse({
-      posts: [
-        {
-          id: 1,
-          name: 'dhh lol',
-        },
-        {
-          id: 2,
-          name: 'james mickens is rad',
-        },
-        {
-          id: 3,
-          name: 'in the name of love',
-        },
-      ],
-    });
+        server.put('/posts/1', function() {
+          return [200, { 'Content-Type': 'application/json' }, null];
+        });
 
-    return run(() => {
-      return store.findAll('post').then(posts => {
-        assert.equal(get(posts, 'length'), 3);
-        posts.forEach(post => assert.ok(post instanceof DS.Model));
+        let post = store.push({ data: { id: '1', type: 'post' } });
+        return post.save().then(() => assert.ok(true, 'save fullfills correctly'));
       });
-    });
+    }
   });
 
-  test('createRecord - sideloaded records are pushed to the store', function(assert) {
-    Post.reopen({
-      comments: DS.hasMany('comment'),
-    });
-
-    ajaxResponse({
-      post: {
-        id: 1,
-        name: 'The Parley Letter',
-        comments: [2, 3],
-      },
-      comments: [
-        {
-          id: 2,
-          name: 'First comment',
-        },
-        {
-          id: 3,
-          name: 'Second comment',
-        },
-      ],
-    });
-    let post;
+  module('init adapter coalesceFindRequests', function() {
+    test('findMany - passes buildURL the requestType', function(assert) {
+      this.owner.register('adapter:application', RESTAdapter.extend({ coalesceFindRequests: true }));
```

Comment:
I think I might just move this to a unit test.  This test file was sort of a pain to work with...

1.  We need to call unregister to register a new adapter with the adapter declaration shown above.
2.  Calling `store.adapterFor` in beforeEach makes it hard to unregister and register a new application adapter since the store `_adapterCache` isn't cleared unless I force clear that private field - which doesn't seem like a good idea.  The approach I took here was to declare two module blocks.  

Still...a unit test is probably the way to go.